### PR TITLE
Refactoring tests

### PR DESCRIPTION
**Pull request by KimberlyMunoz**
Definitely want a second or third pair of 👀 on these.

Using [Owning A Home](https://github.com/cfpb/owning-a-home) as an example, I replaced mocha with mocha_instanbul to give us coverage statistics and wrote tests for more functions in Helpers.js. 

### Other Changes
+ Replaced expect.js with chai to get better errors for easier debugging
+ Replaced the HTML Mocha runner using `compiled_tests.js` with one on the command line that looks at one spec file at a time, again for easier debugging.

### Review
+ @ascott1 
+ @imuchnik
+ @contolini

### CODE REVIEW

**Comment by ascott1** [comment]
:heart: this. I can't wait to dig in.

**Comment by ascott1** [comment]
@KimberlyMunoz @imuchnik: I am running the app, pulled down @KimberlyMunoz's branch, and ran npm install, but when I try to run `grunt mocha_istanbul` I get:

```
0 passing (364ms)
1 pending
2 failing
```

The errors seem related to a "before all" hook.

Did I  miss a step?

Thanks!

**Comment by KimberlyMunoz** [comment]
I just deleted my node_modules folder and ran npm install and didn't get that error. I'm going to check to see if there's a missing file somewhere that didn't get added to this pull request.

**Comment by ascott1** [comment]
Whoops. Looks like I was running an old version of Node*. Using Node 0.12.0 seems to solve that problem.

*[This](http://stackoverflow.com/a/28025834) recommends adding `source $(brew --prefix nvm)/nvm.sh` to your .bash_profile when installing nvm via homebrew. Apparently nvm was not running for me after a recent machine reboot.

**Comment by ascott1** [comment]
This seems to cause issues with routing for me:

screen shot 2015-03-23 at 4 16 15 pm

That URL should be something more like `partial/1005-3/2014-20681`. If I had to guess, I'd say it's related to the change on [this line](https://github.com/cfpb/regulations-site/pull/696/files#diff-666a49b058aa6aed0c62939ec98890e8R129), but I haven't dug in more closely yet.

**Comment by KimberlyMunoz** [comment]
Fixed.

**Comment by KimberlyMunoz** [comment]
🔨 
That should fix it.

File: regulations/static/regulations/js/source/helpers.js [hunk 1]
```diff
@@ -76,20 +75,20 @@ if (!Array.prototype.indexOf) {
     //
     // **Returns** Reg entity marker formatted for human readability
     idToRef: function(id) {
-        var ref = '',
-            parts, i, len, dividers, item, interpIndex, interpParts, subpartIndex;
+        var ref = '';
+        var parts, i, len, dividers, item, interpIndex, interpParts, subpartIndex;
```
Comment (ascott1) [inline regulations/static/regulations/js/source/helpers.js]: This is a small little change, but is such a nice one for readability.

File: regulations/static/regulations/js/source/resources.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+'use strict';
```
Comment (ascott1) [inline regulations/static/regulations/js/source/resources.js]: It's not super clear to me on first blush what a resources file does. Can we add some descriptive comments? Will this only be for this versionElements object? If so, should we rename it `version-resources' or something similar?

**Comment by ascott1** [comment]
This is great and works beautifully. I have one comment above about the naming of the resources file, but once we talk through that, this is ready to merge. Nice work @KimberlyMunoz! And thanks for your help @imuchnik & @contolini!

File: regulations/static/regulations/js/source/resources.js (hunk 2 above)
Comment (KimberlyMunoz) [inline regulations/static/regulations/js/source/resources.js]: I'll add some comments. The file is to pull out the pieces that need to be mocked in tests. 
I named it resources.js so more objects can be added for other functions that need to be refactored, but I'm not opposed to changing the name or doing this in a different way.

Comment (KimberlyMunoz) [inline regulations/static/regulations/js/source/resources.js]: Comments added!

Comment (ascott1) [inline regulations/static/regulations/js/source/resources.js]: Woot woot!

**Pull request merged**